projects/organization-management/src/app/pages/users-create/users-create-page.component.ts

```diff
@@ -0,0 +1,122 @@
+import { Locale } from 'ish-core/models/locale/locale.model';
+import { User } from 'ish-core/models/user/user.model';
+import { whenTruthy } from 'ish-core/utils/operators';
+import { determineSalutations, markAsDirtyRecursive } from 'ish-shared/forms/utils/form-utils';
+
+import { OrganizationManagementFacade } from '../../facades/organization-management.facade';
+
+@Component({
+  selector: 'ish-users-create-page',
+  templateUrl: './users-create-page.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class UsersCreatePageComponent implements OnInit {
+  users$: Observable<User[]>;
+  error$: Observable<HttpError>;
+  loading$: Observable<boolean>;
+  currentLocale$: Observable<Locale>;
+  currentCustomer$: Observable<Customer>;
+
+  titles: string[];
+  currentCountryCode = '';
+  form: FormGroup;
+  submitted = false;
+
+  constructor(
+    private fb: FormBuilder,
+    private organizationManagementFacade: OrganizationManagementFacade,
+    private appFacade: AppFacade
+  ) {}
+
+  ngOnInit() {
+    this.currentCustomer$ = this.organizationManagementFacade.customer$;
+    this.users$ = this.organizationManagementFacade.users$();
+    this.error$ = this.organizationManagementFacade.usersError$;
+    this.loading$ = this.organizationManagementFacade.usersLoading$;
+    this.currentLocale$ = this.appFacade.currentLocale$;
+
+    // determine default language from session and available locales
+    this.currentLocale$.pipe(whenTruthy(), take(1)).subscribe(locale => {
+      this.currentCountryCode = locale.lang.slice(3);
+      this.titles = locale.lang ? determineSalutations(this.currentCountryCode) : undefined;
+    });
+
+    this.createAddUserForm();
+  }
+
+  createAddUserForm() {
+    this.form = this.fb.group({
+      profile: this.fb.group({
+        title: ['', [Validators.required]],
+        firstName: ['', [Validators.required]],
+        lastName: ['', [Validators.required]],
+        login: ['', [Validators.required, CustomValidators.email]],
+        email: ['', [Validators.required, CustomValidators.email]],
+        department: [''],
+        phone: [''],
+        fax: [''],
+        birthday: [''],
+        preferredLanguage: ['en_US', [Validators.required]],
+      }),
+      rolesPermissions: this.fb.group({
+        roles: [''],
+        permissions: [''],
+      }),
+      budgetLimits: this.fb.group({
+        orderSpentLimit: [''],
+        budget: [''],
+      }),
+    });
+  }
+
+  get profile() {
+    return this.form.get('profile');
+  }
+
+  get rolesPermissions() {
+    return this.form.get('rolesPermissions');
+  }
+
+  get budgetLimits() {
+    return this.form.get('budgetLimits');
+  }
+
+  submitForm() {
+    if (this.form.invalid) {
+      this.submitted = true;
+      markAsDirtyRecursive(this.form);
+      return;
+    }
+
+    const formValue = this.form.value;
+    const businessPartnerUsername = formValue.profile.firstName.charAt(0) + formValue.profile.lastName;
```

Comment:
I believe this should be a number but currently the api returns -000001 if I send it empty